packages/mobile/src/components/drawers/ArtistPickConfirmationDrawer.tsx

```diff
@@ -0,0 +1,62 @@
+import { useCallback, useMemo } from 'react'
+
+import { useGetCurrentUser } from '@audius/common/api'
+import { tracksSocialActions, useArtistPickModal } from '@audius/common/store'
+import { useDispatch } from 'react-redux'
+
+import { ConfirmationDrawer } from './ConfirmationDrawer'
+
+const messages = {
+  add: {
+    header: 'Set your Artist Pick',
+    description:
+      'This track will appear at the top of your profile, above your recent uploads, until you change or remove it.',
+    confirm: 'Set Track'
+  },
+  update: {
+    header: 'Change your artist pick?',
+    description:
+      'This track will appear at the top of your profile and replace your previously picked track.',
+    confirm: 'Change Track'
+  },
+  remove: {
+    header: 'Unset as Artist Pick',
+    description:
+      'Are you sure you want to remove your pick? This track will be displayed based on its release date.',
+    confirm: 'Unset Track'
+  }
+}
+
+const { setArtistPick, unsetArtistPick } = tracksSocialActions
+
+export const ArtistPickConfirmationDrawer = () => {
+  const { data } = useArtistPickModal()
+  const { trackId } = data
+  const { data: user } = useGetCurrentUser()
+
+  const dispatch = useDispatch()
+
+  const action = useMemo(
+    () => (!user?.artist_pick_track_id ? 'add' : trackId ? 'update' : 'remove'),
```

Comment:
I think that I'm missing context here. Will trackId be passed sometimes and if it is passed, would that be remove or update?

Reply:
yeah its a bit odd. what i noticed is that when you set the artist pick, as the drawer is closing user.artist_pick_track_id gets set which changes the action, resulting in it going from "add" to "remove" as it's sliding down. the memo ensures we only have one action through the lifecycle of the drawer.